sdk/core/allocator/revoker.h

```diff
@@ -52,6 +52,18 @@ namespace Revocation
 			shadowCap = const_cast<WordT *>(MMIO_CAPABILITY(WordT, shadow));
 		}
 
+		WordT shadow_paint_bits_above(ptraddr_t addr)
```

Comment:
Minor pedantry, but I don't really like the names of these two.  `paint` is both a verb and a noun (and we have a style guide rule to avoid those words!) and it took me a couple of reads of the function to realise that you meant the noun and not the verb form.  Maybe `shadow_bits_below_address`?